packages/ethereum/contracts/test/Channels.t.sol

```diff
@@ -0,0 +1,1441 @@
+      HoprChannels.Balance.wrap(uint96(amount1)),
+      HoprChannels.TicketIndex.wrap(0),
+      HoprChannels.Timestamp.wrap(0),
+
+      HoprChannels.ChannelEpoch.wrap(1),
+      HoprChannels.ChannelStatus.OPEN
+    );
+
+    vm.prank(address(1));
+    vm.mockCall(
+      vm.addr(1),
+      abi.encodeWithSignature(
+        'transferFrom(address,address,uint256)',
+        address(1),
+        address(hoprChannels),
+        amount1 + amount2
+      ),
+      abi.encode(true)
+    );
+
+    vm.expectEmit(true, false, false, false, address(hoprChannels));
+    emit ChannelOpened(accountA.accountAddr, accountB.accountAddr, HoprChannels.Balance.wrap(amount1));
+
+    vm.expectEmit(true, false, false, false, address(hoprChannels));
+    emit ChannelOpened(accountB.accountAddr, accountA.accountAddr, HoprChannels.Balance.wrap(amount2));
+    
+    hoprChannels.fundChannelMulti(accountA.accountAddr, HoprChannels.Balance.wrap(amount1), accountB.accountAddr, HoprChannels.Balance.wrap(amount2));
+    vm.clearMockedCalls();
+
+    vm.expectEmit(true, false, false, true, address(hoprChannels));
+    emit CommitmentSet(channelIdAB, HoprChannels.ChannelEpoch.wrap(1));
+
+    vm.prank(accountB.accountAddr);
+    hoprChannels.setCommitment(bytes32(commitment1), accountA.accountAddr);
+
+    assertEq(
+      keccak256(
+        abi.encode(getChannelFromTuple(channelIdAB))
+      ),
+      keccak256(abi.encode(channelAB))
+    );
+
+    vm.prank(accountB.accountAddr);
+    vm.expectRevert(abi.encodeWithSelector(InvalidCommitment.selector));
+    hoprChannels.setCommitment(bytes32(commitment1), accountA.accountAddr);
+
+    HoprChannels.Channel memory channelABBumped = HoprChannels.Channel(
+      bytes32(commitment2),
+      HoprChannels.Balance.wrap(uint96(amount1)),
+      HoprChannels.TicketIndex.wrap(0),
+      HoprChannels.Timestamp.wrap(0),
+
+      HoprChannels.ChannelEpoch.wrap(2),
+      HoprChannels.ChannelStatus.OPEN
+    );
+
+
+    vm.expectEmit(true, false, false, true, address(hoprChannels));
+    emit CommitmentSet(channelIdAB, HoprChannels.ChannelEpoch.wrap(2));
+
+    vm.prank(accountB.accountAddr);
+    hoprChannels.setCommitment(bytes32(commitment2), accountA.accountAddr);
+
+    assertEq(
+      keccak256(
+        abi.encode(getChannelFromTuple(channelIdAB))
+      ),
+      keccak256(abi.encode(channelABBumped))
+    );
+  }
+
+  function testFailSetCommitment() public {
+    vm.prank(accountA.accountAddr);
+    vm.expectRevert(abi.encodeWithSelector(InvalidCommitment.selector, ""));
+    hoprChannels.setCommitment(bytes32(0), accountB.accountAddr);
+  }
+
+  function testFailSetCommitmentChannelNotOpen(uint256 commitment) public {
+    commitment = bound(commitment, uint256(1), type(uint256).max);
+
+    vm.prank(accountA.accountAddr);
+    vm.expectRevert(abi.encodeWithSelector(WrongChannelState.selector, ""));
+    hoprChannels.setCommitment(bytes32(commitment), accountB.accountAddr);
+  }
+
+  // function testOpenChannel
```

Comment:
What are those commented-out test cases? If they are not WIP nor needed then please remove

Reply:
they are "to be migrated in next PR". work has not finished. this is just an intermediate PR